src/middleware/operation.rs

```diff
@@ -476,23 +484,31 @@ fn check_custom_pred(
         }
     }
 
-    let wildcard_map = match resolve_wildcard_values(params, pred, args) {
-        Some(wc_map) => wc_map,
-        None => return Ok(false),
-    };
+    let wildcard_map = resolve_wildcard_values(params, pred, args)?;
 
-    // Check that the resolved wildcard match the statement arguments.
+    // Check that the resolved wildcards match the statement arguments.
     for (s_arg, wc_value) in s_args.iter().zip(wildcard_map.iter()) {
         if *wc_value != *s_arg {
-            return Ok(false);
+            return Err(Error::mismatched_wildcard_value_and_statement_arg(
```

Comment:
Maybe we can add the `pred` here and also the index of the argument for better context?